### ISSUE TITLE
compose scale web=50 fails

### ISSUE THREAD

**Issue by aanm** [docker/compose#1977]
I think parallel scale should be more resilient. I tried to do `compose scale web=50` and I saw a couple of messages like this one:
```
Exception in thread Thread-4:
Traceback (most recent call last):
  File "/code/build/docker-compose/out00-PYZ.pyz/threading", line 810, in __bootstrap_inner
  File "/code/build/docker-compose/out00-PYZ.pyz/threading", line 763, in run
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.utils", line 31, in inner_execute_function
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 222, in <lambda>
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 175, in create_and_start
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 276, in create_container
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.container", line 42, in create
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 245, in create_container
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 252, in create_container_from_config
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.clientbase", line 128, in _post_json
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.clientbase", line 83, in _post
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 508, in post
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 465, in request
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 573, in send
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.adapters", line 415, in send
ConnectionError: ('Connection aborted.', ResponseNotReady())
```

Since scale has a `-t` timeout argument I think it would make sense to have a `-r` retries argument that retries N more times.

**Comment by aanand** [comment]
Yep - two things going on here. First, that *particular* exception should be fixed by https://github.com/docker/compose/pull/1974.

Second, `scale` (as well as `stop`, `kill` and `rm`) now handle errors much better, thanks to https://github.com/docker/compose/pull/1879.

As for retrying: might be nice to have, but you can always just run the command again.

**Comment by aanm** [comment]
@aanand Cool! I'm also seeing:

```
Exception in thread Thread-8:
Traceback (most recent call last):
  File "/code/build/docker-compose/out00-PYZ.pyz/threading", line 810, in __bootstrap_inner
  File "/code/build/docker-compose/out00-PYZ.pyz/threading", line 763, in run
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.utils", line 31, in inner_execute_function
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 222, in <lambda>
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 176, in create_and_start
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.container", line 140, in start
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.utils.decorators", line 20, in wrapped
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 797, in start
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.clientbase", line 128, in _post_json
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.clientbase", line 83, in _post
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 508, in post
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 465, in request
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 573, in send
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.adapters", line 433, in send
ReadTimeout: HTTPConnectionPool(host='localhost', port=2375): Read timed out. (read timeout=60)
```
With `-t 1000`, is that also fixed by #1974 and #1879?

**Comment by aanand** [comment]
We're going to show a better error for timeouts with https://github.com/docker/compose/pull/1963, but you raise a good point - if the user has passed an explicit timeout to an operation with `-t`, we should also respect that when waiting for the HTTP request to complete.

**Comment by dnephin** [comment]
I believe this is a duplicate of, or will be fixed by the same change as #1938 , so going to close as a duplicate.
